.circleci/config.yml

```diff
@@ -59,30 +90,45 @@ jobs:
       - run: wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
       - run: sudo dpkg -i google-chrome-stable_current_amd64.deb
       - run: git submodule update --init
-      - run: yarn
+      - restore_cache:
```

Comment:
is it needed to restore cache in the `oracle-e2e` target as well?

Reply:
It's not, because:
- In `ui-e2e` we install dependencies in a CI step, so it's worth to cache yarn package cache
- In `oracle-e2e` we don't install any dependencies in a CI step (only in the context of a docker build)

The difference comes from the fact that `oracle-e2e` tests are run in a docker container, whereas the `ui-tests` selenium tests need to run in a [machine executor](https://circleci.com/docs/2.0/executor-types/#using-machine)

Reply:
Hmm... I see.

Does it mean that caching could be implemented when https://github.com/poanetwork/tokenbridge/issues/57 is addressed? Will it speed up CI checks?

Reply:
Not exactly because in `ui-e2e`, the `UI` already does run in a docker container (although using `Dockerfile.e2e` and not dedicated production Docker image like `oracle`).

What needs to be run in a machine executor is the headless chrome + selenium tests, which are accessing UI exposed from the docker container.

Having run the `ui-e2e` tests many times on my machine, I noticed that waiting for confirmations is taking a large portion of the time. Maybe we should create a separate issue to try and reduce required confirmations / speed up block mining?

Reply:
Right, the `ui` waits for 8 block confirmations even if the `oracle` processed the transactions after waiting 1 block as are configured on deployed contracts for `ui-e2e` tests. There is an issue created for this on the old bridge-ui repo https://github.com/poanetwork/bridge-ui/issues/119 to wait for the configured value on the contracts

Reply:
I will move the issues from the standalone repos on this week.